dns/dns_updater.go

```diff
@@ -242,6 +246,15 @@ func (u *updater) createChanges(hostToIngress hostToIngress,
 	return changes, skipped
 }
 
+func findMatchingRecord(host string, originalRecords []*route53.ResourceRecordSet) (bool, *route53.ResourceRecordSet) {
+	for _, recordSet := range originalRecords {
+		if host == *recordSet.Name {
+			return true, recordSet
+		}
+	}
+	return false, nil
+}
```

Comment:
Do you think it would be better if we used a map of the original records instead of having to iterate through them each time? We could create a map with key of host + elb DNS name. I think it would simplify the logic a bit too.